petastorm/spark/spark_dataset_converter.py

```diff
@@ -139,19 +176,58 @@ def __len__(self):
         """
         return self.dataset_size
 
-    def make_tf_dataset(self):
+    def make_tf_dataset(
+            self,
+            batch_size=32,
+            prefetch=None,
+            num_epochs=None,
+            workers_count=None,
+            **petastorm_reader_kwargs
+    ):
         """
         Make a tensorflow dataset.
 
         This method will do the following two steps:
           1) Open a petastorm reader on the materialized dataset dir.
           2) Create a tensorflow dataset based on the reader created in (1)
 
+        :param batch_size: The number of items to return per batch
+        :param prefetch: Prefetch size for tensorflow dataset. If None will use
+            tensorflow autotune size. Note only available on tensorflow>=1.14
+        :param num_epochs: An epoch is a single pass over all rows in the dataset.
+            Setting ``num_epochs`` to ``None`` will result in an infinite number
+            of epochs.
+        :param workers_count: An int for the number of workers to use in the
+            reader pool. This only is used for the thread or process pool.
+            None denotes auto tune best value (current implementation when auto tune,
+            it will always use 4 workers, but it may be improved in future)
+            Default value None.
+        :param petastorm_reader_kwargs: arguments for `petastorm.make_batch_reader()`,
+            exclude these arguments: "dataset_url", "num_epochs", "workers_count".
+
         :return: a context manager for a `tf.data.Dataset` object.
                  when exit the returned context manager, the reader
                  will be closed.
         """
-        return TFDatasetContextManager(self.cache_dir_url)
+
+        # override some arguments default values of petastorm reader
+        petastorm_reader_kwargs['num_epochs'] = num_epochs
+        if workers_count is None:
+            # TODO: generate a best tuned value for default worker count value
+            workers_count = 4
+        petastorm_reader_kwargs['workers_count'] = workers_count
+
+        if not is_rank_and_size_consistent_with_env(petastorm_reader_kwargs):
+            logger.warning('The petastorm arguments cur_shard and shard_count is not '
```

Comment:
Find it hard to understand this message. Can we add more information here? Perhaps show current petastorm sharding settings and horovod ones and suggest a course of action to fix the issue.

Do you mean `each work sample` -> `each worker sample`?

If it does sample whole dataset, is that a problem?